src/main/java/controller/UserController.java

```diff
@@ -0,0 +1,257 @@
+package controller;
+
+import enums.Gender;
+import model.common.Address;
+import model.user.impl.User;
+import repository.UserRepository;
+
+import java.util.regex.Pattern;
+
+
+/**
+ * Creates connection between the user and the user repository.
+ * Provides authentication and validation of the user.
+ * Operates on the user repository.
+ */
+public class UserController {
+
+    private final UserRepository userRepository;
+
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    /**
+     * Provides a check if a user with that username is already registered and
+     * if user exists validates the provided password.
+     *
+     * @param username - username provided by the user.
+     * @param password - password provided by the user.
+     * @return true if authentication is successful or false on failure.
+     */
+    public boolean validateCredentials(String username, String password) {
+        User userToAuthenticate = getUser(username);
```

Comment:
Instead of, using this method `getUser(username)`, you can use `getUserByUsername()`. I think it's doing the same